problems/CHBP/chbp.chpl

```diff
@@ -0,0 +1,220 @@
+module CHBP {
+  // Character-Based Phylogen
+  // http://rosalind.info/problems/chbp/
+  const defaultfilename="none";
+  config const infile=defaultfilename;
+
+  const empty_set: domain(string);
+
+  // use stdin if filename == defaultfilename
+  iter lines_from_file(filename: string, defaultfilename: string) throws {
+    var channel = stdin;
+
+    if infile != defaultfilename {
+      var file = open(infile, iomode.r);
+      channel = file.reader();
+    }
+
+    var line: string;
+    while (channel.readline(line)) do
+      yield line.strip();
+  }
+
+
+  class Partition {
+    // A partition of a list of labels, generated by either a set of labels
+    // or a charactable entry (a string of labels.size 0s and 1s)
+    var set, complement: domain(string);
+
+    proc init(labels, char_table_line:string) {
+      const universe : domain(string) = labels;
+      set = empty_set; // this is absurd but I don't see how else to initialize the empty set.
```

Comment:
@ljdursi - I follow this project and happened to notice this. You should be able to work around the lack of a empty associative domain literal by using a [`this.complete()`](https://chapel-lang.org/docs/master/_downloads/chapelLanguageSpec.pdf#page=138), which will default-initialize any uninitialized fields at that point.